src/js/components/Shared/TracksList.react.js

```diff
@@ -189,158 +124,224 @@ export default class TracksList extends Component {
-                        this.setState({ selected });
-                        break;
-                    }
-                    case 1: {
-
-                        const onlySelected = selected[0];
-                        let onlySelectedIndex;
-
-                        for(let i = 0, length = tracks.length; i < length; i++) { // STH wrong here
-                            if(tracks[i]._id === onlySelected) {
-                                onlySelectedIndex = i;
-                                break;
-                            }
-                        }
-
-                        if(index < onlySelectedIndex) {
-                            for(let i = 1; i <= Math.abs(index - onlySelectedIndex); i++) {
-                                selected.push(tracks[onlySelectedIndex - i]._id);
-                            }
-                        } else if(index > onlySelectedIndex) {
-                            for(let i = 1; i <= Math.abs(index - onlySelectedIndex); i++) {
-                                selected.push(tracks[onlySelectedIndex + i]._id);
-                            }
-                        }
-
-                        self.setState({ selected });
-                        break;
-                    }
-                    default: {
-                        const selectedInt = [];
-
-                        for(let i = 0, length = tracks.length; i < length; i++) {
-                            if(selected.includes(tracks[i]._id)) {
-                                selectedInt.push(i);
-                            }
-                        }
-
-                        let base;
-                        const min = Math.min(...selectedInt);
-                        const max = Math.max(...selectedInt);
-
-                        if(index < min) {
-                            base = max;
-                        } else {
-                            base = min;
-                        }
-
-                        const newSelected = [];
-
-                        if(index < min) {
-                            for(let i = 0; i <= Math.abs(index - base); i++) {
-                                newSelected.push(tracks[base - i]._id);
-                            }
-                        } else if(index > max) {
-                            for(let i = 0; i <= Math.abs(index - base); i++) {
-                                newSelected.push(tracks[base + i]._id);
-                            }
-                        }
-
-                        self.setState({ selected : newSelected });
-                        break;
-                    }
+        return tracksChunked.splice(tilesScrolled, tilesToDisplay).map((tracksChunk, indexChunk) => {
+
+            const list = tracksChunk.map((track, index) => {
+
+                let playingIndicator = undefined;
+
+                if(trackPlayingId === track._id) {
+                    playingIndicator = <PlayingIndicator state={ Player.getAudio().paused ? 'pause' : 'play' } />;
```

Comment:
can we move it to a separate method? cause now it has if+ternary, that makes it hard to read